scripts/curate_basel.py

```diff
@@ -0,0 +1,127 @@
+    dict_images = {
+        "MP2RAGE_UNI_Images.nii.gz": "_UNIT1.nii.gz"
+    }
+
+    dict_der = {
+        "lesion_mask_sc.nii.gz": "_UNIT1_lesion-manual.nii.gz"
+    }
+
+    for dirs, subdirs, files in os.walk(root_data):
+        for file in files:
+            if file.endswith('.nii.gz') and file in dict_images.keys() or file in dict_der.keys():
+                path_file_in = os.path.join(dirs, file)
+                path = os.path.normpath(path_file_in)
+                subid_bids = 'sub-' + (path.split(os.sep))[5]
+                flag_der = False
+                if file.endswith('lesion_mask_sc.nii.gz'):
+                    path_subid_bids_dir_out = os.path.join(output_data, 'derivatives', 'labels', subid_bids, 'anat')
+                    flag_der = True
+                else:
+                    path_subid_bids_dir_out = os.path.join(output_data, subid_bids, 'anat')
+                if not os.path.isdir(path_subid_bids_dir_out):
+                    os.makedirs(path_subid_bids_dir_out)
+                if flag_der is False:
+                    path_file_out = os.path.join(path_subid_bids_dir_out, subid_bids + dict_images[file])
+                else:
+                    path_file_out = os.path.join(path_subid_bids_dir_out, subid_bids + dict_der[file])
+                shutil.copy(path_file_in, path_file_out)
+
+        for dirName, subdirList, fileList in os.walk(output_data):
+            for file in fileList:
+                if file.endswith('.nii.gz'):
+                    originalFilePath = os.path.join(dirName, file)
+                    jsonSidecarPath = os.path.join(dirName, file.split(".")[0] + '.json')
+                    if os.path.exists(jsonSidecarPath) is False:
+                        print("Missing: " + jsonSidecarPath)
+                        if file.endswith('lesion-manual.nii.gz'):
+                            data_json_label = {}
+                            data_json_label[u'Author'] = "USB"
+                            data_json_label[u'Label'] = "lesion-manual"
+                            with open(jsonSidecarPath, 'w') as outfile:
+                                outfile.write(json.dumps(data_json_label, indent=2, sort_keys=True))
+                            outfile.close()
+                        else:
+                            os.system('touch ' + jsonSidecarPath)
+
+    sub_list = os.listdir(output_data)
+    sub_list.remove('derivatives')
+
+    sub_list.sort()
+
+    import csv
+
+    participants = []
+    for subject in sub_list:
+        row_sub = []
+        row_sub.append(subject)
+        row_sub.append('n/a')
+        row_sub.append('n/a')
+        participants.append(row_sub)
+
+    print(participants)
+    with open(output_data + '/participants.tsv', 'w') as tsv_file:
+        tsv_writer = csv.writer(tsv_file, delimiter='\t', lineterminator='\n')
+        tsv_writer.writerow(["participant_id", "sex", "age"])
+        for item in participants:
+            tsv_writer.writerow(item)
```

Comment:
You can make this more efficient by streaming it, like using a unix pipe:

```suggestion
    with open(output_data + '/participants.tsv', 'w') as tsv_file:
        tsv_writer = csv.writer(tsv_file, delimiter='\t', lineterminator='\n')
        tsv_writer.writerow(["participant_id", "sex", "age"])
        for subject in sub_list:
            row_sub = [subject, 'n/a', 'n/a']
            print(row_sub)
            tsv_writer.writerow(row_sub)
```

I know this is probably irrelevant to this dataset, but on huge datasets this technique is the difference between feasible and not :basketball: